input/netmap/netmap.c

```diff
@@ -623,28 +922,360 @@ static int catch_audio(struct uprobe *uprobe, struct upipe *upipe,
+
+    UBASE_SIGNATURE_CHECK(args, UPIPE_PROBE_UREF_SIGNATURE);
+    struct uref *uref = va_arg(args, struct uref *);
+    va_arg(args, struct upump **);
+    va_arg(args, bool *);
+
+    const struct sdi_picture_fmt *fmt = NULL;
+    for (int i = 0; i < sizeof(pict_fmts) / sizeof(*pict_fmts); i++) {
+        if (pict_fmts[i].interlaced != netmap_opts->interlaced)
+            continue;
+        if (pict_fmts[i].active_height != netmap_opts->height)
+            continue;
+        fmt = &pict_fmts[i];
+        break;
+    }
+
+    if (!fmt) {
+        upipe_err(upipe, "Unknown format");
+        return UBASE_ERR_INVALID;
+    }
+
+    size_t hsize, vsize, stride;
+    const uint8_t *r;
+    if (unlikely(!ubase_check(uref_pic_size(uref, &hsize, &vsize, NULL)) ||
+                 !ubase_check(uref_pic_plane_size(uref, "x10", &stride,
+                                                  NULL, NULL, NULL)) ||
+                 !ubase_check(uref_pic_plane_read(uref, "x10", 0, 0, -1, -1,
+                                                  &r)))) {
+        upipe_throw_error(upipe, UBASE_ERR_INVALID);
+        return UBASE_ERR_INVALID;
+    }
+
+    for (unsigned int i = 0; i < vsize; i++) {
+        r += stride;
+
+        const uint16_t *x = (const uint16_t*)r;
+
+        size_t h_left = hsize;
+        while (h_left > S291_HEADER_SIZE + S291_FOOTER_SIZE) {
+            if (x[0] != S291_ADF1 || x[1] != S291_ADF2 || x[2] != S291_ADF3) {
+                break;
+            }
+
+            uint8_t did = s291_get_did(x);
+            uint8_t sdid = s291_get_sdid(x);
+            uint8_t dc = s291_get_dc(x);
+            if (S291_HEADER_SIZE + dc + S291_FOOTER_SIZE > h_left) {
+                upipe_dbg_va(upipe, "ancillary too large (%"PRIu8" > %zu) for 0x%"PRIx8"/0x%"PRIx8,
+                        dc, h_left, did, sdid);
+                break;
+            }
+
+            if (!s291_check_cs(x)) {
+                upipe_dbg_va(upipe, "invalid CRC for 0x%"PRIx8"/0x%"PRIx8,
+                        did, sdid);
+                x += 3;
+                h_left -= 3;
+                continue;
+            }
+
+            if (!send_did_sdid(netmap_ctx, did, sdid)) {
+                x += 3;
+                h_left -= 3;
+                continue;
+            }
+
+            obe_coded_frame_t *anc_frame = netmap_ctx->anc_frame;
+            if (!anc_frame) {
+                netmap_ctx->anc_frame = anc_frame = new_coded_frame(output_stream->output_stream_id, 65536);
+                if (!anc_frame)
+                    return UBASE_ERR_ALLOC;
+
+                anc_frame->len = 0;
+            }
+
+            bool c_not_y = false;
+
+            bs_t s;
+            bs_init(&s, &anc_frame->data[anc_frame->len], 65536 - anc_frame->len);
+
+            bs_write(&s, 6, 0);
+            bs_write(&s, 1, c_not_y);
+            bs_write(&s, 11, vanc_line_number(fmt, i));
+            bs_write(&s, 12, hsize - h_left + get_vanc_offset(netmap_ctx));
```

Comment:
This offset is relative to SAV, get_vanc_offset seems to be relative to EAV

Reply:
Is it ?

Reply:
It doesn't make much but it seems so

Reply:
How would you write HANC on line 1, horizontal_offset = 1920+16/2 and line_number = 1125 on previous frame?